File: test-suite/src/test/java/io/micronaut/docs/http/server/response/textplain/TextPlainControllerTest.java
package io.micronaut.docs.http.server.response.textplain;

import io.micronaut.context.annotation.Property;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;
import java.util.Calendar;

import static org.junit.jupiter.api.Assertions.assertEquals;

@Property(name = "spec.name", value = "TextPlainControllerTest")
@MicronautTest
class TextPlainControllerTest {

    @Client("/txt")
    @Inject
    HttpClient httpClient;

    @Test
    void textPlainBoolean() {
        asserTextResult("/boolean", "true");
    }

    @Test
    void textPlainMonoBoolean() {
        asserTextResult("/boolean/mono", "true");
    }

    @Test
    void textPlainFluxBoolean() {
        asserTextResult("/boolean/flux", "true");
    }

    @Test
    void textPlainBigDecimal() {
        asserTextResult("/bigdecimal", BigDecimal.valueOf(Long.MAX_VALUE).toString());
    }

    @Test
    void textPlainDate() {
        asserTextResult("/date", new Calendar.Builder().setDate(2023,7,4).build().toString());
    }

    @Test
    void textPlainPerson() {
        asserTextResult("/person", new Person("Dean Wette", 65).toString());
    }

    private void asserTextResult(String url, String expectedResult) {
        final String result = httpClient.toBlocking().retrieve(url);
        assertEquals(expectedResult, result);

    }
}

File: test-suite-python/src/test/python/micronaut/docs/http/server/response/textplain/TextPlainControllerTest.py
from typing import Annotated

import java
from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .Person import Person

BigDecimal = java.type("java.math.BigDecimal")
Calendar = java.type("java.util.Calendar")
Long = java.type("java.lang.Long")


@Property(name="spec.name", value="TextPlainControllerTest")
@MicronautTest
class TextPlainControllerTest:
    httpClient: Annotated[HttpClient, Inject, Client("/txt")]

    @Test
    def text_plain_boolean(self):
        self.assert_text_result("/boolean", "true")

    @Test
    def text_plain_mono_boolean(self):
        self.assert_text_result("/boolean/mono", "true")

    @Test
    def text_plain_flux_boolean(self):
        self.assert_text_result("/boolean/flux", "true")

    @Test
    def text_plain_big_decimal(self):
        self.assert_text_result("/bigdecimal", BigDecimal.valueOf(Long.MAX_VALUE).toString())

    @Test
    def text_plain_date(self):
        self.assert_text_result("/date", Calendar.Builder().setDate(2023, 7, 4).build().toString())

    @Test
    def text_plain_person(self):
        self.assert_text_result("/person", str(Person("Dean Wette", 65)))

    def assert_text_result(self, url: str, expected_result: str):
        result = self.httpClient.toBlocking().retrieve(url)
        assert result == expected_result
